CollapseLauncher/Classes/RepairManagement/Zenless/ZenlessRepair.Fetch.cs

```diff
@@ -0,0 +1,467 @@
+            UpdateStatus();
+
+            // Initialize new proxy-aware HttpClient
+            using HttpClient client = new HttpClientBuilder()
+                .UseLauncherConfig(_downloadThreadCount + _downloadThreadCountReserved)
+                .SetUserAgent(_userAgent)
+                .SetAllowedDecompression(DecompressionMethods.None)
+                .Create();
+
+            // Initialize the new DownloadClient
+            DownloadClient downloadClient = DownloadClient.CreateInstance(client);
+
+            // Create a hash set to overwrite local files
+            Dictionary<string, FilePropertiesRemote> hashSet = new Dictionary<string, FilePropertiesRemote>(StringComparer.OrdinalIgnoreCase);
+
+            // If not in cache mode, then fetch main package
+            if (!IsCacheUpdateMode)
+            {
+                // Get the primary manifest
+                await GetPrimaryManifest(hashSet, token, assetIndex);
+            }
+
+            // Execute on non-recover main mode
+            if (!IsOnlyRecoverMain)
+            {
+                // Get the in-game res manifest
+                await GetResManifest(downloadClient, hashSet, token, assetIndex,
+#if DEBUG
+                    true
+#else
+                    false
+#endif
+                    );
+
+                // Execute plugin things if not in cache mode only
+                if (!IsCacheUpdateMode)
+                {
+                    // Force-Fetch the Bilibili SDK (if exist :pepehands:)
+                    await FetchBilibiliSDK(token);
+
+                    // Remove plugin from assetIndex
+                    // Skip the removal for Delta-Patch
+                    EliminatePluginAssetIndex(assetIndex);
+                }
+            }
+        }
+        #endregion
+
+        #region PrimaryManifest
+        private async Task GetPrimaryManifest(Dictionary<string, FilePropertiesRemote> hashSet, CancellationToken token, List<FilePropertiesRemote> assetIndex)
+        {
+            // If it's using cache update mode, then return since we don't need to add manifest
+            // from pkg_version on cache update mode.
+            if (IsCacheUpdateMode)
+                return;
+
+            // Initialize pkgVersion list
+            List<PkgVersionProperties> pkgVersion = new List<PkgVersionProperties>();
+
+            // Initialize repo metadata
+            try
+            {
+                // Get the metadata
+                Dictionary<string, string> repoMetadata = await FetchMetadata(token);
+
+                // Check for manifest. If it doesn't exist, then throw and warn the user
+                if (!(repoMetadata.TryGetValue(_gameVersion.VersionString, out var value)))
+                {
+                    throw new VersionNotFoundException($"Manifest for {_gameVersionManager.GamePreset.ZoneName} (version: {_gameVersion.VersionString}) doesn't exist! Please contact @neon-nyan or open an issue for this!");
```

Comment:
/Q
Can we fallback directly to game metadata's PkgVersion or is there a difference in ours that directly using the other causes issue?

Reply:
Pushed